### PR TITLE
Common clinical cases and treatment features

### PR DESCRIPTION
**Pull request by labkey-martyp**
#### Rationale
Copy some commonly used clinical and treatment data entry and reporting into EHR.

#### Related Pull Requests
* https://github.com/LabKey/johnsHopkinsEHRModules/pull/35

#### Changes
* Add EHRShowEditWithoutFormUIButton and VetReviewButton
* Add treatmentSchedule query, views and metadata
* Add ActiveCasesDemographicProvider and several views for viewing cases
* Add AnimalGroups add and remove clinical history data sources
* Add ExamCasesDataEntryPanel
* Add medication, diet, surgical and incomplete schedules
* Update base reports to include Daily Reports with cases and treatment reports

### CODE REVIEW

File: ehr/resources/queries/study/treatmentSchedule.query.xml [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+                        </item>
+                        <item text="Return Distinct Values">
+                            <onClick>
+                                EHR.window.GetDistinctWindow.getDistinctHandler(dataRegionName);
+                            </onClick>
+                        </item>
+                    </item>
+                    <item text=" " requiresSelection="true" insertAfter="More Actions">
+                        <onClick>javascript:void(0);</onClick>
+                    </item>
```
Comment (labkey-jeckels) [inline ehr/resources/queries/study/treatmentSchedule.query.xml]: This seems odd. A button item with no text and no handler?

File: ehr/resources/web/ehr/panel/ExamCasesDataEntryPanel.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+           }, this);
+
+           if (missingCases.length){
+               Ext4.Msg.confirm('No Case', 'There is no active clinical case for this animal, do you want to continue anyway?', function(val){
+                   if (val == 'yes'){
+                       this.onSubmit(btn, true);
+                   }
+                   else {
+
+                   }
```
Comment (labkey-jeckels) [inline ehr/resources/web/ehr/panel/ExamCasesDataEntryPanel.js]:
```suggestion
```

**Review by labkey-jeckels — APPROVED** [review]
This may add a lot of new views for the cases dataset for all the centers - do you think they'll generally work?

And are you planning to remove any copies of these files from ONPRC or elsewhere?

File: ehr/resources/queries/study/treatmentSchedule.query.xml (hunk 1 above)
Comment (bbimber) [reply ehr/resources/queries/study/treatmentSchedule.query.xml]: the children are injected by table customizers. there's a registerMoreActionsButton() on EHRService or something to that effect. 

I cant remember the code, but it's possible the java customize code does (or should) simply create this button if it doesnt exist. I dont recall why this is done here.

Comment (labkey-martyp) [reply ehr/resources/queries/study/treatmentSchedule.query.xml]: I couldn't find anything using this so I just removed it for now

File: ehr/resources/web/ehr/panel/ExamCasesDataEntryPanel.js (hunk 2 above)
Comment (labkey-martyp) [reply ehr/resources/web/ehr/panel/ExamCasesDataEntryPanel.js]: fixed

**Pull request merged**